Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,144 @@ public struct Incident: Codable, Equatable {
         case geometryIndexStart = "geometry_index_start"
         case geometryIndexEnd = "geometry_index_end"
     }
-
-    var identifier: String
-    var type: String
-    var description: String
-    var creationTime: String
-    var startTime: String
-    var endTime: String
-    var impact: String?
-    var subtype: String?
-    var subtypeDescription: String?
-    var alertCodes: [Int]
-    var lanesBlocked: [Int]
-    var geometryIndexStart: Int
-    var geometryIndexEnd: Int
+    
+    /// Defines known types of incidents.
+    ///
+    /// Each incident may or may not have specific set of data, depending on it's `type`
+    public enum IncidentType: String {
+        case Accident = "accident"
+        case Congestion = "congestion"
+        case Construction = "construction"
+        case DisabledVehicle = "disabled_vehicle"
+        case LaneRestriction = "lane_restriction"
+        case MassTransit = "mass_transit"
+        case Miscellaneous = "miscellaneous"
+        case OtherNews = "other_news"
+        case PlannedEvent = "planned_event"
+        case RoadClosure = "road_closure"
+        case RoadHazard = "road_hazard"
+        case Weather = "weather"
+        
+        case Unknown = "<<unknown type>>"
+    }
+    
+    /// Incident identifier
+    public var identifier: String
+    /// The kind of an incident
+    public var type: IncidentType
+    var rawType: String
+    /// Short description of an incident. May be used as an additional info.
+    public var description: String
+    /// Date when incident item was created. Uses ISO8601 format
+    public var creationDate: Date
+    /// Date when incident happened. Uses ISO8601 format
+    public var startDate: Date
+    /// Date when incident shall end. Uses ISO8601 format
+    public var endDate: Date
+    /// Shows severity of an incident. May be not available for all incident types.
+    public var impact: String?
+    /// Provides additional classification of an incident. May be not available for all incident types.
+    public var subtype: String?
+    /// Breif description of the subtype. May be not available for all incident types and is not available if `subtype` is `nil`
+    public var subtypeDescription: String?
+    /// Contains list of ISO 14819-2:2013 codes
+    ///
+    /// See https://www.iso.org/standard/59231.html for details
+    public var alertCodes: [Int]
```

Comment:
Does the order matter? If not, maybe it could be represented by a Set. (JSON doesn’t have a native set type, hence the use of an array.)

Reply:
Done